### PR TITLE
ref(js): spring cleaning

### PR DESCRIPTION
**Pull request by JonasBa**
I have been experimenting with some eslint rules to detect performance issues as per https://github.com/getsentry/sentry-javascript/pull/7910 and https://github.com/getsentry/sentry/pull/47629. All of this is pretty much in the same spirit where we try to find On^2 code + unnecessary array rest/spread.

Figured it wouldn't hurt to clean some of the code in Sentry anyways so I'm opening this as draft and awareness. I would not expect any of these to be actual performance issues, but the code is still wasteful and in some cases just less readable (like the ...set.length vs just checking set.size)

Lmk what you think, I'd be curious to hear some thoughts and opinions, the only thing to cleanup is just the chang to eslintrc which was linked for testing purposes

### CODE REVIEW

File: static/app/views/issueList/actions/actionSet.tsx [hunk 1]
```diff
@@ -59,9 +59,15 @@ function ActionSet({
...
-    .filter(issue => issue) as BaseGroup[];
+  const selectedIssues: BaseGroup[] = [];
+  for (const issue of issues) {
+    const resolvedIssue = GroupStore.get(issue);
+
+    if (resolvedIssue) {
+      // @TODO use a type guard here
+      selectedIssues.push(resolvedIssue as BaseGroup);
+    }
+  }
```
Comment (scttcper) [inline static/app/views/issueList/actions/actionSet.tsx:70]: this feels harder to read

Comment (scttcper) [reply static/app/views/issueList/actions/actionSet.tsx:70]: could just switch it to issues.map so there isn't an extra array being made

File: static/app/views/issueList/actions/index.tsx [hunk 2]
```diff
@@ -273,10 +273,15 @@ function useSelectedGroupsState() {
...
-    .map(group => group.project.slug);
+  const projects: string[] = [];
+
+  for (const selectedGroup of selected) {
+    const group = GroupStore.get(selectedGroup);
+
+    if (group && group.project) {
+      projects.push(group.project.slug);
+    }
+  }
```
Comment (scttcper) [inline static/app/views/issueList/actions/index.tsx:284]: i feel like the map and filter should just be written better

maybe
```ts
const projects = selected
    .map(id => GroupStore.get(id)?.project.slug)
    .filter(defined)
```

File: static/app/components/forms/fields/choiceMapperField.tsx [hunk 3]
```diff
@@ -125,7 +125,10 @@ export default class ChoiceMapperField extends Component<ChoiceMapperFieldProps>
     } = props;
 
     const mappedKeys = Object.keys(columnLabels);
-    const emptyValue = mappedKeys.reduce((a, v) => ({...a, [v]: null}), {});
+    const emptyValue = mappedKeys.reduce((a, v) => {
+      a[v] = null;
+      return a;
+    }, {});
```
Comment (kamilogorek) [inline static/app/components/forms/fields/choiceMapperField.tsx:131]: Alternatively: 
```suggestion
    const emptyValue = Object.fromEntries(mappedKeys.map(key => ([key, null])));
```

File: static/app/components/forms/fields/tableField.tsx [hunk 4]
```diff
@@ -56,7 +56,13 @@ export default class TableField extends Component<InputFieldProps> {
     } = props;
 
     const mappedKeys = columnKeys || [];
-    const emptyValue = mappedKeys.reduce((a, v) => ({...a, [v]: null}), {id: ''});
+    const emptyValue = mappedKeys.reduce(
```
Comment (kamilogorek) [inline static/app/components/forms/fields/tableField.tsx:59]: Same as above, but with `.concat(['id', ''])` at the end of `map`.

File: static/app/actionCreators/pageFilters.tsx [hunk 5]
```diff
@@ -234,7 +234,9 @@ export function initializeUrlState({
       // the organization slug is consistent with the URL param--Sentry will
       // get the first project from the organization that the user is a member
       // of.
-      newProject = [...memberProjects].slice(0, 1).map(getProjectIdFromProject);
+      newProject = memberProjects?.[0]
+        ? [getProjectIdFromProject(memberProjects[0])]
+        : [];
```
[the remark this answers is not in the document]
Comment (JonasBa) [reply static/app/actionCreators/pageFilters.tsx:239]: I think what happens is that calling memberProjects with undefined throws (least from looking at the test failures), so I'd need to protect from that

File: static/app/components/forms/fields/choiceMapperField.tsx (hunk 3 above)
Comment (JonasBa) [reply static/app/components/forms/fields/choiceMapperField.tsx:131]: Oh nice, I like that. I think we had some errors around Object.fromEntries in unsupported browsers though, I'd need to check

File: static/app/views/issueList/actions/actionSet.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/views/issueList/actions/actionSet.tsx:70]: Actually, set.forEach is better here, same with below comment I think

File: static/app/views/issueList/actions/index.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/views/issueList/actions/index.tsx:284]: It does not work because selected is a Set :/

**Pull request closed without merging**